src/app.h

```diff
@@ -0,0 +1,384 @@
+                    cv::cvtColor(image, image, cv::COLOR_RGBA2BGR);
+                    cv::imshow("Pitch result", pitch);
+                    cv::imshow("Image result", image);
+                    cv::waitKey(0);
+                    cv::destroyAllWindows();
+                } catch (const std::exception& ex) {
+                    logging::error("{}", ex.what());
+                }
+            })
+            .item("Reset", [this] {
+                _model.reset();
+                renderImages();
+            });
+
+        _gui.main_menu().button("Debug", [this] { _state.debugWindowFlag = not _state.debugWindowFlag; });
+    }
+
+    void renderWindows() {
+        if (_state.loadPopupFlag) {
+            _gui.popup("Load")
+                .input("path", &_model.imagePath())
+                .separator()
+                .button("Ok", vec2{ 120, 0 }, [this] {
+                    _model.load();
+
+                    glGenTextures(1, &_state.imageTexture);
+                    glBindTexture(GL_TEXTURE_2D, _state.imageTexture);
+                    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+                    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+
+                    // Set texture clamping method
+                    const auto& image = _model.image();
+                    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP);
+                    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP);
+                    glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, image.cols, image.rows, 0, GL_RGBA, GL_UNSIGNED_BYTE, image.ptr());
+
+                    _state.showImage = true;
+                    _state.loadPopupFlag = false;
+                    ImGui::CloseCurrentPopup();
+                })
+                .focus()
+                .same_line()
+                .button("Cancel", vec2{ 120, 0 }, [this] {
+                    _state.loadPopupFlag = false;
+                    ImGui::CloseCurrentPopup();
+                })
+                .open();
+        }
+
+        if (_state.savePopupFlag) {
+            _gui.popup("Save")
+                .input("path", &_state.savePath)
+                .separator()
+                .button("Ok", vec2{ 120, 0 }, [this] {
+                    _model.save(_state.savePath);
+                    _state.savePopupFlag = false;
+                    ImGui::CloseCurrentPopup();
+                })
+                .focus()
+                .same_line()
+                .button("Cancel", vec2{ 120, 0 }, [this] {
+                    _state.savePopupFlag = false;
+                    ImGui::CloseCurrentPopup();
+                })
+                .open();
+        }
+
+        if (_state.debugWindowFlag) {
+            _gui.window("Debug", ui::window_config{ ui::window_flag::None }, &_state.debugWindowFlag)
+                .text("Logic time: {:.3f}ms", static_cast<double>(_delta.count()) / 1'000'000.0)
```

Comment:
No "naked" `count()`s. Prefer explicit casting before using `count()`.
For an example: https://github.com/GK72/abyss/commit/4e7221d870829da7eb0d683f86cf7f0522417b3e#diff-ca54668860ba1b6ad285983b8a53a637120a3c906f4f124f0a974f864d1bd4caR112